src/main/java/org/commcare/formplayer/services/RestoreFactory.java

```diff
@@ -622,8 +632,8 @@ private HttpHeaders getHmacHeaders(String requestPath) {
         if( asUsername != null) {
             builder.append("&as=").append(asUsername).append("@").append(domain).append(".commcarehq.org");
         }
-        if (skipFixtures) {
-            builder.append("&skip_fixtures=True");
+        if (skipFixtures && skipFixturesAfterSubmit()) {
```

Comment:
I think for conciseness / consistency it would be good to move this check upstream to the original request in the Form Controller